src/test/java/org/ahocorasick/trie/StateTest.java

```diff
@@ -19,6 +22,52 @@ public void constructSequenceOfCharacters() {
         assertEquals(2, currentState.getDepth());
         currentState = currentState.nextState('c');
         assertEquals(3, currentState.getDepth());
+        currentState = currentState.nextState('F');
+        assertNull(currentState);
```

Comment:
Is there a way to avoid `null` here? Perhaps the `nextState` should return `Optional`?

Reply:
I don't think, it's a good idea, it's an internal API, the reason I added all these extra unit tests, just for the sake of improving the code coverage.
Code coverage has dropped probably because I removed `ListElementRemoval.java` and its unit test altogether.